### PR TITLE
Repair byte length determination problem

### PR DESCRIPTION
**Pull request by Adsicmes**
Its a bug fix about [This issue #4 ](https://github.com/jaasonw/osu-db-tools/issues/4)

### CODE REVIEW

File: buffer.py [hunk 1]
```diff
@@ -79,18 +79,18 @@ def write_ulong(self, data: int):
         self.data += struct.pack("<Q", data)
 
     def write_string(self, data: str):
-        if (len(data) > 0):
+        if len(data) > 0:
             self.write_ubyte(0x0b)
             strlen = b""
-            value = len(data)
+            value = len(data.encode('utf-8'))
```
Comment (Adsicmes) [inline buffer.py:85]: Get the byte length of the characters instead of the characters length.

Comment (jaasonw) [reply buffer.py:85]: oh shoot, thanks for looking into it for me, you've tested this already?

Comment (Adsicmes) [reply buffer.py:85]: Yeah. I can show my test code if you need

Comment (jaasonw) [reply buffer.py:85]: oh alright, it's fine then, one of these days i'll have to write up some unit tests LOL

**Comment by Adsicmes** [comment]
[二进制读写尝试.zip](https://github.com/jaasonw/osu-db-tools/files/9437385/default.zip)
Maybe you can try this.
Its a very simple test for it. Even if I use some Chinese annotation.
Unzip this zip will show the code. Run test.py.

**Pull request merged**